plugins/CoreUpdater/Diagnostic/HttpsUpdateCheck.php

```diff
@@ -31,8 +32,11 @@ public function execute()
     {
         $label = $this->translator->translate('Installation_SystemCheckUpdateHttps');
 
-        if (CoreUpdater\Controller::isUpdatingOverHttps()) {
+        if (Http::isUpdatingOverHttps() || GeneralConfig::getConfigValue('force_matomo_ssl_request') === 0) {
             return array(DiagnosticResult::singleResult($label, DiagnosticResult::STATUS_OK));
+        } else {
+            // if check failed, set force_matomo_ssl_request to 0
+            GeneralConfig::setConfigValue('force_matomo_ssl_request', 0);
```

Comment:
We don't want to automatically set this to 0, we want to show a `DiagnosticResult::STATUS_WARNING` with the information about setting this config value to 0